src/Lido/contracts/CrossChainExecutor.sol

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import {IBaseReceiverPortal} from '../../contracts/interfaces/IBaseReceiverPortal.sol';
+
+import {BridgeExecutorBase} from './BridgeExecutorBase.sol';
+
+/**
+ * @title CrossChainExecutor
+ * @author Lido
+ * @notice Contract that implements receiver portal role along with the ability to execute actions sets.
+ * @dev Queuing an ActionsSet into this Executor can only be done by sending a message from Ethereum Governance Agent
+ * via a.DI CrossChainController contract on the Ethereum mainnet using this contract as a destination.
+ */
+contract CrossChainExecutor is BridgeExecutorBase, IBaseReceiverPortal {
+
+  /**
+   * @dev Address of the CrossChainController contract on the current chain.
+   */
+  address private immutable CROSS_CHAIN_CONTROLLER;
+
+  /**
+   * @dev Address of the DAO Agent contract on the root chain.
+   */
+  address private immutable GOVERNANCE_EXECUTOR;
```

Comment:
not sure if should and can be immutable though

Reply:
Can you explain why in a bit more detail?

I thought we agreed that any change to the executor would trigger a new executor to be deployed and vote to switch to this new executor. Should we consider moving the executor under the prox so as not to update all related contracts in case of executor replacement?